### ISSUE TITLE
inversion model training time and loss

### ISSUE THREAD

**Issue by nayammy** [vec2text/vec2text#89]
Hi, thank you for the great work and open-sourcing the code.

I'm currently training the inversion model using a single H100 GPU, and I wanted to check if the training behavior I'm seeing is expected.

The estimated training time is around 90 hours.

At around 50 epochs, the loss is approximately 1.3.

I'm using the following training configuration:
`--per_device_train_batch_size 1024 \
--per_device_eval_batch_size 1024 \
--max_seq_length 32 \
--model_name_or_path t5-base \
--dataset_name nq \
--embedder_model_name gtr_base \
--num_repeat_tokens 16 \
--embedder_no_grad True \
--num_train_epochs 100 \
--max_eval_samples 500 \
--eval_steps 40000 \
--warmup_steps 25000 \
--bf16=1 \
--use_wandb=1 \
--use_frozen_embeddings_as_input True \
--experiment inversion \
--lr_scheduler_type linear \
--exp_group_name oct-gtr \
--learning_rate 0.001 \
--output_dir ./saves/gtr-1024-2 \
--save_steps 8000`

My questions are:

Does a training time of ~90 hours on a single H100 sound reasonable for this setup?

Is a loss of around 1.3 at 50 epochs expected?

When using CLIP as the embedding model instead of GTR, I noticed that the loss is significantly worse. Is this behavior expected due to CLIP’s embedding characteristics?

Any insights or guidance would be greatly appreciated. Thanks again!

**Comment by jxmorris12** [comment]
This sounds about right to me in both loss and training time.

> When using CLIP as the embedding model instead of GTR, I noticed that the loss is significantly worse.

This is a great question. I don't know the specifics of CLIP but it makes sense to me. First of all, CLIP has smaller embeddings; second of all, they're trained through a noisier image-text contrastive process which is likely less precise in some semantic sense than GTR's text-only training. So I am not surprised. Would be interested to see apples-to-apples analysis here.

If your main concern is performance you could try [Pile T5](https://huggingface.co/EleutherAI/pile-t5-large/blob/main/config.json) which is larger and better. Use the biggest size you can. It likely has some performance optimizations as well.
